Pcap++/src/PcapFileDevice.cpp

```diff
@@ -114,6 +118,93 @@ int IFileReaderDevice::getNextPackets(RawPacketVector& packetVec, int numOfPacke
 	return numOfPacketsRead;
 }
 
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// SnoopFileReaderDevice members
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+SnoopFileReaderDevice::~SnoopFileReaderDevice()
+{
+	m_snoopFile.close();
+}
+
+bool SnoopFileReaderDevice::open()
+{
+	m_NumOfPacketsRead = 0;
+	m_NumOfPacketsNotParsed = 0;
+
+	m_snoopFile.open(m_FileName.c_str(), std::ifstream::binary);
+	snoop_file_header_t snoop_file_header;
+	m_snoopFile.read((char*)&snoop_file_header, sizeof(snoop_file_header_t));
+
+	if(be64toh(snoop_file_header.identification_pattern) != 0x736e6f6f70000000 && be32toh(snoop_file_header.version_number) == 2)
+		return false;
+
+	// From https://datatracker.ietf.org/doc/html/rfc1761
+	static const pcpp::LinkLayerType snoop_encap[] = {
+		LINKTYPE_ETHERNET,	/* IEEE 802.3 */
+		LINKTYPE_NULL,		/* IEEE 802.4 Token Bus */
+		LINKTYPE_IEEE802_5,	/* IEEE 802.5 */
+		LINKTYPE_NULL,		/* IEEE 802.6 Metro Net */
+		LINKTYPE_ETHERNET,	/* Ethernet */
+		LINKTYPE_C_HDLC,	/* HDLC */
+		LINKTYPE_NULL,		/* Character Synchronous, e.g. bisync */
+		LINKTYPE_NULL,		/* IBM Channel-to-Channel */
+		LINKTYPE_FDDI		/* FDDI */
+	};
+
+	m_PcapLinkLayerType = snoop_encap[be32toh(snoop_file_header.datalink_type)];

```

Comment:
We can check that the datalink type is in range:
```cpp
if (snoop_file_header.datalink_type > 9)
{
	PCPP_LOG_ERROR("Cannot read data link type for '" << m_FileName << "');
	m_snoopFile.close();
	return false;
}
```

Reply:
Resolved in 2e24bdf